browser/net/brave_ad_block_tp_network_delegate_helper.cc

```diff
@@ -75,17 +176,18 @@ void OnBeforeURLRequestAdBlockTP(
   }
   DCHECK_NE(ctx->request_identifier, 0UL);
 
-  g_brave_browser_process->ad_block_service()->GetTaskRunner()
-      ->PostTaskAndReply(FROM_HERE,
-                         base::BindOnce(&ShouldBlockAdOnTaskRunner, ctx),
-                         base::BindOnce(&OnShouldBlockAdResult, next_callback,
-                                        ctx));
-}
+  scoped_refptr<base::SequencedTaskRunner> task_runner =
+      g_brave_browser_process->ad_block_service()->GetTaskRunner();
 
-int OnBeforeURLRequest_AdBlockTPPreWork(
-    const ResponseCallback& next_callback,
-    std::shared_ptr<BraveRequestInfo> ctx) {
+  base::OnceCallback<void(base::Optional<std::string>)> cname_callback =
+      base::BindOnce(&ShouldBlockAdWithOptionalCname, task_runner,
```

Comment:
I don't understand why do we need this callback. Can we just call `ShouldBlockAdWithOptionalCname` in `OnComplete`?
`task_runner` will be alive given that it is owned by `ad_block_service` which in turn is owned by `g_browser_process`.

Or we can bind an anonymous lambda here that would take `get task_runner` and post

Reply:
Good call, we needed this previously in the (now removed) failure case when we couldn't create the resolver

Reply:
I moved it into the `AdblockCnameResolveHostClient` constructor since you discovered we need to bind the `ResponseCallback` anyways.